src/transformStateWithClones.js

```diff
@@ -7,7 +7,49 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateArray = [];
+
+  // stateArray.push({ ...state });
+
+  // console.log(newState, actions);
+  let last = { ...state };
+
+  for (const el of actions) {
+    switch (el.type) {
```

Comment:
Use desctructuring here and add more descriptive name
```suggestion
  for (const action of actions) {
    const { type, ...... } = action;
  
    switch (type) {
```